src/core/ext/transport/chttp2/transport/frame_settings.h

```diff
@@ -46,12 +46,35 @@ struct grpc_chttp2_settings_parser {
   uint16_t id;
   uint32_t value;
 };
-// Create an ack settings frame
+
+// Create an ack settings frame as per RFC
+// https://datatracker.ietf.org/doc/html/rfc9113#name-settings-format
 grpc_slice grpc_chttp2_settings_ack_create(void);
 
+// The function has two tasks
+// 1. Initialise the parser with information that will be needed later, at the
+//    time of parsing the entire settings frame.
+// 2. Running checks to see if the settings frame content is valid or not.
+//    Returns an error with a helpful error string if something is wrong.
+//
+// TODO(tjagtap) : [PH2][P2] : Use ABSL_MUST_USE_RESULT for return value.
 grpc_error_handle grpc_chttp2_settings_parser_begin_frame(
     grpc_chttp2_settings_parser* parser, uint32_t length, uint8_t flags,
     grpc_core::Http2Settings& settings);
+
+// Mandatory to call grpc_chttp2_settings_parser_begin_frame calling before
+// calling grpc_chttp2_settings_parser_parse. However our code does not assert
+// that yet.
+// This function parses the settings frame. A lot of the parsed content is saved
+// in the parser object itself.
+// This function may return different errors, be sure to check them.
+//
+// TODO(tjagtap) : [PH2][P2] : Assert that that
```

Comment:
for ph2 we should be using the api in `frame.h` and not this older api

One of the major problems that our original frame/deframe code has (in `frame_*.{h,cc}` but not `frame.{h,cc}`) is that it mixes functionality (how to react to settings frames) with serialization.

Reply:
Got it.... But the new stuff in frame.h is not being used inside any transport right?

Reply:
Removed the comment